src/account.ts

```diff
@@ -1,23 +1,113 @@
-import * as ethjsUtil from 'ethjs-util'
+import * as assert from 'assert'
+import * as BN from 'bn.js'
+import * as rlp from 'rlp'
+import { stripHexPrefix } from 'ethjs-util'
+import { KECCAK256_RLP, KECCAK256_NULL } from './constants'
+import { zeros, bufferToHex, toBuffer } from './bytes'
+import { keccak, keccak256, keccakFromString, rlphash } from './hash'
+import { assertIsHexString, assertIsBuffer } from './helpers'
+import { BNLike, BufferLike, bnToRlp } from './types'
+
 const {
   privateKeyVerify,
   publicKeyCreate,
   publicKeyVerify,
   publicKeyConvert,
 } = require('ethereum-cryptography/secp256k1')
-import * as assert from 'assert'
-import * as BN from 'bn.js'
-import { zeros, bufferToHex } from './bytes'
-import { keccak, keccak256, keccakFromString, rlphash } from './hash'
-import { assertIsHexString, assertIsBuffer } from './helpers'
 
-/**
- * Returns a zero address.
- */
-export const zeroAddress = function(): string {
-  const addressLength = 20
-  const addr = zeros(addressLength)
-  return bufferToHex(addr)
+export interface AccountData {
+  nonce?: BNLike
+  balance?: BNLike
+  stateRoot?: BufferLike
+  codeHash?: BufferLike
+}
+
+export class Account {
+  nonce: BN
+  balance: BN
+  stateRoot: Buffer
+  codeHash: Buffer
+
+  static fromAccountData(accountData: AccountData) {
+    const { nonce, balance, stateRoot, codeHash } = accountData
+
+    return new Account(
+      nonce ? new BN(toBuffer(nonce)) : undefined,
+      balance ? new BN(toBuffer(balance)) : undefined,
+      stateRoot ? toBuffer(stateRoot) : undefined,
+      codeHash ? toBuffer(codeHash) : undefined,
+    )
+  }
+
+  public static fromRlpSerializedAccount(serialized: Buffer) {
+    const values = rlp.decode(serialized)
+
+    if (!Array.isArray(values)) {
+      throw new Error('Invalid serialized account input. Must be array')
+    }
+
+    return this.fromValuesArray(values)
+  }
+
+  public static fromValuesArray(values: Buffer[]) {
+    const [nonce, balance, stateRoot, codeHash] = values
+
+    return new Account(
+      nonce ? new BN(nonce) : undefined,
+      balance ? new BN(balance) : undefined,
+      stateRoot,
+      codeHash,
```

Comment:
Why do `stateRoot` and `codeHash` don't have optionality here in contrast to `nonce` and `balance`?

Reply:
The BNs need to be wrapped in `new BN` to convert from buffer to BN. If the hashes are not provided they will default to undefined and the constructor will take care of the default values.

Reply:
Ah, makes sense.